lib/spack/spack/cmd/mirror.py

```diff
@@ -143,9 +143,22 @@ def setup_parser(subparser):
     )
 
 
+def promote_deprecated_path_to_url(path_or_url):
+    # Spack tutorial still mentions spack mirror add <name> <path>
+    # but the commands really expect URLs.
+    if not url_util.is_path_instead_of_url(path_or_url):
+        return path_or_url
+
+    url = url_util.path_to_file_url(path_or_url)
+
+    tty.warn("Converted path to URL: {}.".format(url))
```

Comment:
I don't think this should be a warning. If you're using a filesystem mirror, you should be able to supply a path, right?  Is the issue that on windows we're unsure of whether we're doing this correctly?  `is_path_instead_of_url` looked pretty good to me... are there situations where it'll still do something unexpected and we need to warn?

If not, I think this can either be a debug message, or we can remove it.

Reply:
Hm, not sure to be honest. We should just improve the API. If we promote without warning you get weird things like:

`spack mirror add x binaries.spack.io/develop` => `file://binaries.spack.io/develop`

If we expect a URL we can error.

Reply:
I also don't like `spack mirror add [--directory ... | --mirror-url ...]` to be honest :sweat:

Reply:
The documentation is quite explicit that it requires a URL, so let's just stick to the warning.

```
$ spack mirror add --help
usage: spack mirror add ... mirror url

positional arguments:
  mirror                mnemonic name for mirror
  url                   url of mirror directory from 'spack mirror create'
```